In [1]:
import pandas as pd

In [15]:
potholes = pd.read_csv('CSV/Potholes_Patched.csv')

In [16]:
potholes.columns


Index(['CREATION DATE', 'STATUS', 'COMPLETION DATE', 'SERVICE REQUEST NUMBER',
       'TYPE OF SERVICE REQUEST', 'CURRENT ACTIVITY', 'MOST RECENT ACTION',
       'NUMBER OF POTHOLES FILLED ON BLOCK', 'STREET ADDRESS', 'ZIP',
       'X COORDINATE', 'Y COORDINATE', 'Ward', 'Police District',
       'Community Area', 'LATITUDE', 'LONGITUDE', 'LOCATION'],
      dtype='object')

In [17]:
potholes.index

RangeIndex(start=0, stop=40, step=1)

In [18]:
potholes.describe()

NUMBER OF POTHOLES FILLED ON BLOCK           ZIP  X COORDINATE  \
count                           40.000000     40.000000  4.000000e+01   
mean                            11.650000  60620.450000  1.162390e+06   
std                             14.130436     14.507204  6.111527e+03   
min                              1.000000  60607.000000  1.153826e+06   
25%                              3.000000  60612.000000  1.157479e+06   
50%                              6.500000  60612.000000  1.162469e+06   
75%                             15.000000  60622.000000  1.166134e+06   
max                             63.000000  60651.000000  1.179091e+06   

       Y COORDINATE       Ward  Police District  Community Area   LATITUDE  \
count  4.000000e+01  40.000000        40.000000       40.000000  40.000000   
mean   1.902942e+06  25.800000        12.375000       24.950000  41.889386   
std    4.376646e+03   6.169403         2.349986        4.477579   0.012081   
min    1.895655e+06   1.000000         1.000000        8.000000  41.869371   
25%    1.898743e+06  26.000000        12.000000       24.000000  41.877815   
50%    1.902627e+06  27.000000        12.000000       25.500000  41.888501   
75%    1.905939e+06  27.000000        12.000000       28.000000  41.897684   
max    1.910450e+06  42.000000        18.000000       28.000000  41.909986   

       LONGITUDE  
count  40.000000  
mean  -87.679251  
std     0.022419  
min   -87.711021  
25%   -87.697323  
50%   -87.679252  
75%   -87.665553  
max   -87.617560

In [32]:
potholes_police_district = potholes[['NUMBER OF POTHOLES FILLED ON BLOCK', 'Police District']]

In [33]:
potholes_police_district.describe()

NUMBER OF POTHOLES FILLED ON BLOCK  Police District
count                           40.000000        40.000000
mean                            11.650000        12.375000
std                             14.130436         2.349986
min                              1.000000         1.000000
25%                              3.000000        12.000000
50%                              6.500000        12.000000
75%                             15.000000        12.000000
max                             63.000000        18.000000

On an average, 11 potholes are filled.

In [36]:
potholes_police_district['Police District'].value_counts()

12    30
14     7
18     2
1      1
Name: Police District, dtype: int64

Police district number 12 has 30 potholes filled. 

In [42]:
potholes_community_area = potholes[['NUMBER OF POTHOLES FILLED ON BLOCK', 'Community Area']]

In [43]:
potholes_community_area['Community Area'].value_counts()

28    17
24    11
23     7
27     3
8      2
Name: Community Area, dtype: int64

In [41]:
potholes_community_area.loc[potholes_community_area['Community Area']== 28]

NUMBER OF POTHOLES FILLED ON BLOCK  Community Area
1                                   15              28
3                                   19              28
6                                    3              28
7                                   12              28
8                                    3              28
9                                    1              28
10                                  25              28
21                                   8              28
22                                   4              28
24                                  18              28
25                                   6              28
27                                   3              28
29                                   2              28
32                                  19              28
33                                  63              28
35                                   7              28
36                                  11              28

Community Area, 28, has filled the largest number of potholes.

In [45]:
df = potholes_community_area.loc[potholes_community_area['Community Area']== 28]

In [46]:
df['NUMBER OF POTHOLES FILLED ON BLOCK'].sum()

219

Community 28 has filled 219 potholes.

In [47]:
potholes_community_area.loc[potholes_community_area['Community Area']== 24]['NUMBER OF POTHOLES FILLED ON BLOCK'].sum()

88

The nearest competitor to community 28 is community 24 with 88 potholes filled.

In [48]:
219 - 88

131

How are community 28 managing to fill 131 more potholes than its nearest competitor? Other communities can learn from their practices.